src/webgpu/shader/validation/expression/call/builtin/select.spec.ts

```diff
@@ -0,0 +1,232 @@
+    }
+  })
+  .fn(t => {
+    const type = kArgumentTypes[t.params.type];
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      /* expectedResult */ type === Type.bool,
+      [Type.i32.create(0), Type.i32.create(0), type.create(0)],
+      'constant',
+      /*return_type*/ Type.i32
+    );
+  });
+
+const kTests = {
+  valid: {
+    src: `_ = ${builtin}(1, 2, true);`,
+    pass: true,
+  },
+  alias: {
+    src: `_ = ${builtin}(i32_alias(1), i32_alias(2), bool_alias(true));`,
+    pass: true,
+  },
+  bool: {
+    src: `_ = ${builtin}(false, false, true);`,
+    pass: true,
+  },
+  i32: {
+    src: `_ = ${builtin}(1i, 1i, true);`,
+    pass: true,
+  },
+  u32: {
+    src: `_ = ${builtin}(1u, 1u, true);`,
+    pass: true,
+  },
+  f32: {
+    src: `_ = ${builtin}(1.0f, 1.0f, true);`,
+    pass: true,
+  },
+  f16: {
+    src: `_ = ${builtin}(1.0h, 1.0h, true);`,
+    pass: true,
+  },
+  mixed_aint_afloat: {
+    src: `_ = ${builtin}(1, 1.0, true);`,
+    pass: true,
+  },
+  mixed_i32_u32: {
+    src: `_ = ${builtin}(1i, 1u, true);`,
+    pass: false,
+  },
+  vec_bool: {
+    src: `_ = ${builtin}(vec2<bool>(false, true), vec2<bool>(false, true), true);`,
+    pass: true,
+  },
+  vec_i32: {
+    src: `_ = ${builtin}(vec2<i32>(1, 1), vec2<i32>(1, 1), true);`,
+    pass: true,
+  },
+  vec_u32: {
+    src: `_ = ${builtin}(vec2<u32>(1, 1), vec2<u32>(1, 1), true);`,
+    pass: true,
+  },
+  vec_f32: {
+    src: `_ = ${builtin}(vec2<f32>(1, 1), vec2<f32>(1, 1), true);`,
+    pass: true,
+  },
+  vec_f16: {
+    src: `_ = ${builtin}(vec2<f16>(1, 1), vec2<f16>(1, 1), true);`,
+    pass: true,
+  },
+  matrix: {
+    src: `_ = ${builtin}(mat2x2(1, 1, 1, 1), mat2x2(1, 1, 1, 1), true);`,
+    pass: false,
+  },
+  atomic: {
+    src: ` _ = ${builtin}(a, a, true);`,
+    pass: false,
+  },
+  array: {
+    src: `var a: array<bool, 5>;
+            _ = ${builtin}(a, a, true);`,
+    pass: false,
+  },
+  array_runtime: {
+    src: `_ = ${builtin}(k.arry, k.arry, true);`,
+    pass: false,
+  },
+  struct: {
+    src: `var a: A;
+            _ = ${builtin}(a, a, true);`,
+    pass: false,
+  },
+  enumerant: {
+    src: `_ = ${builtin}(read_write, read_write, true);`,
+    pass: false,
+  },
+  ptr: {
+    src: `var<function> a = true;
+            let p: ptr<function, bool> = &a;
+            _ = ${builtin}(p, p, true);`,
+    pass: false,
+  },
+  ptr_deref: {
+    src: `var<function> a = true;
+            let p: ptr<function, bool> = &a;
+            _ = ${builtin}(*p, *p, true);`,
+    pass: true,
+  },
+  sampler: {
+    src: `_ = ${builtin}(s, s, true);`,
+    pass: false,
+  },
+  texture: {
+    src: `_ = ${builtin}(t, t, true);`,
+    pass: false,
+  },
+  no_args: {
+    src: `_ = ${builtin}();`,
+    pass: false,
+  },
+  too_few_args: {
+    src: `_ = ${builtin}(1, true);`,
+    pass: false,
+  },
+  too_many_args: {
+    src: `_ = ${builtin}(1, 1, 1, true);`,
+    pass: false,
+  },
+};
+
+g.test('arguments')
```

Comment:
Please can we have a `must_use` test too?

Reply:
Done. Also added to the list: https://github.com/gpuweb/cts/issues/3613